app/assets/javascripts/admin/addon/mixins/setting-component.js

```diff
@@ -143,7 +216,26 @@ export default Mixin.create({
   settingName: computed("setting.setting", "setting.label", function () {
     const setting = this.setting?.setting;
     const label = this.setting?.label;
-    return label || setting.replace(/\_/g, " ");
+    const name = label || setting.replace(/\_/g, " ");
+
+    const formattedName = (name.charAt(0).toUpperCase() + name.slice(1))
+      .split(" ")
+      .map((word) =>
+        ACRONYMS.has(word.toLowerCase()) ? word.toUpperCase() : word
+      )
+      .map((word) => {
+        if (word.endsWith("s")) {
+          const singular = word.slice(0, -1).toLowerCase();
+          return ACRONYMS.has(singular) ? singular.toUpperCase() + "s" : word;
+        }
+        return word;
+      })
+      .join(" ");
+
+    return MIXED_CASE.reduce(
+      (acc, [key, value]) => acc.replaceAll(new RegExp(key, "gi"), value),
```

Comment:
Should `key` be surrounded with word boundaries? I think at the moment without word boundaries, "photos" for example would become "phoToS" since it would match the "ToS" entry.

Reply:
Good catch, fixed!